carbon-location.html

```diff
@@ -148,8 +148,8 @@
         }
       },
 
-      __computeRoutePath: function(path, hash, useHashAsPath) {
-        return useHashAsPath ? hash : path;
+      __computeRoutePath: function() {
+        return this.useHashAsPath ? this.__hash : this.__path;
```

Comment:
Actually, on further testing, this does affect the re-entrancy conditions. Most consistently on safari and firefox.

Reading the properties directly from `this` fixes it.